### ISSUE TITLE
resolvedExpectedArtifacts not populated when using orca's SQL pipelineRef feature

### ISSUE THREAD

**Issue by dethi** [spinnaker/spinnaker#6993]
### Issue Summary:

orca's SQL pipelineRef feature doesn't populate the resolvedExpectedArtifacts field, causing error when using `#triggerResolvedArtifact(String name)`.

### Cloud Provider(s):

- GCP
- Kubernetes

### Environment:

- Spinnaker 1.35.4
- Using pipeline trigger to propagate between clusters. Pipeline A deploy the Cluster A. Pipeline B has a trigger set to match successful run of pipeline B. Pipeline B match the docker artifacts of Pipeline A to decide which version to deploy.

### Feature Area:

- Pipeline trigger

### Steps to Reproduce:

- Spinnaker 1.35.4, with Orca config: `executionRepository.sql.enabled=true` and `executionRepository.sql.pipelineRef.enabled=true`
- Create two pipeline, pipelineA and pipelineB
- PipelineB has an automated trigger on pipelineA successful execution, with an artifact constraints
- PipelineB has an `Evaluate Variables` stage that use [#triggerResolvedArtifact(String name)](https://spinnaker.io/docs/reference/pipeline/expressions/#triggerresolvedartifactstring-name) to use the expected artifact.
- Trigger pipelineA and see pipelineB triggering successfully but the resolvedExpectedArtifacts field is empty causing the `Evaluate Variables` stage to fail

### Additional Details:

- [pipelineRef](https://github.com/spinnaker/orca/pull/4749)

- Errors:
```
"expressionEvaluationSummary": {
                    "trigger['expectedArtifacts'][0]['boundArtifact']['version']": [
                        {
                            "description": "Failed to evaluate [COMMITID] EL1012E: Cannot index into a null value",
                            "exceptionType": "org.springframework.expression.spel.SpelEvaluationException",
                            "level": "ERROR",
                            "timestamp": 1731919796541
                        }
                    ]
                },
```

- Trigger setup:
<img width="1037" alt="Screenshot 2024-11-18 at 09 42 56" src="https://github.com/user-attachments/assets/dace6c9d-6016-4215-a650-f6bd25bc5216">

**Comment by edgarulg** [comment]
thanks for submitting the bug @dethi. I was able to reproduce the issue and I will be working on a fix for this.

**Comment by edgarulg** [comment]
I raised a PR with the fix for this issue: https://github.com/spinnaker/orca/pull/4816

**Comment by edgarulg** [comment]
@dethi the fix for this issue was merged and backported. We still need to create a new patch release to deliver it. I will close this issue if you agree and you can re-open it if you still see issues.

**Comment by dethi** [comment]
Thanks! I will test again when released.

**Comment by edgarulg** [comment]
@dethi I released a new version of Spinnaker 1.35.5 and it has the fix for this issue.
